File: Python/Robot Name.py
import random
import string

class Robot:
    def __init__(self):
        self.name = Robot.name(self)

    def name(self):
        name = ''.join(random.choice(string.ascii_uppercase) for x in range(2))
        name += ''.join(random.choice(str(random.randint(0, 9))) for x in range(3))
        return name

    def reset(self):
         random.seed(random.randint(0, 1024))
         self.name = Robot.name(self)

print(Robot().name)
print(Robot().name)

File: C#/Robot Name.cs
using System;
using System.Threading;
using System.Text;
using System.Collections.Generic;


public class Robot
{
    public Robot()
    {
        this.Name = NameGen();
    }

    public string Name;

    public static List<string> robotNames = new List<string>();

    public static string letters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";

    public string NameGen()
    {
        Random rnd = new Random();
        StringBuilder sb = new StringBuilder();
        string name = sb.Append(letters[rnd.Next(0, letters.Length)]).Append(letters[rnd.Next(0, letters.Length)]).Append(rnd.Next(0, 9)).Append(rnd.Next(0, 9)).Append(rnd.Next(0, 9)).ToString();
        if (robotNames.Contains(name))
        {
            return NameGen();
        }
        else
        {
            robotNames.Add(name);
            return name;
        }
    }

    public void Reset()
    {
        this.Name = NameGen();
    }
}